index.js

```diff
@@ -214,7 +217,7 @@ function rateLimitRequestHandler (pluginComponent, params) {
       const res = await new Promise((resolve, reject) => {
         store.incr(key, (err, res) => {
           err ? reject(err) : resolve(res)
-        }, max, params.ban)
+        }, timeWindow, max, params.ban)
```

Comment:
This will be a breaking change since you are modifying the signature of the increment method, this method is called with custom store implementations too, so they would need to be modified

Another change that modifies this method's signature is queued up in the next branch ([it removed the ban parameter from incr](https://github.com/fastify/fastify-rate-limit/commit/fe1ca6f2fd7d16f590fb34ca8a48acdc508ad293)), I suggest releasing these changes together in the next major that will support Fastify v5

Reply:
Cool, sounds good! Do you have an idea how far in the future fastify 5 might be?

Reply:
After March… but we started making the major releases for the plugins, so don’t see any reason not to do it here as well

Reply:
Awesome! :)

Reply:
@gurgunday any chance for a realease any time soon?

Reply:
@mcollina should we release `next` like some other plugins?

We'd no longer be testing for node 14 but it's fine I think